services/graph/app/consumer.py

```
@@ -306,7 +306,14 @@ def hybrid_deserializer(value, ctx):
                             # Use string for storage/passing
                             tenant_id = str(tid_obj)
                         except ValueError:
-                            pass
+                            logger.error(
+                                "invalid_tenant_id_in_message",
+                                tenant_id=tenant_id,
+                                doc_id=doc_id,
+                            )
+                            raise ValueError(
+                                f"Invalid tenant_id '{tenant_id}' — refusing to fall back to global database"
+                            )
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)